system/HTTP/ContentSecurityPolicy.php

```diff
@@ -624,17 +584,20 @@ protected function addOption($options, string $target, bool $explicitReporting =
 		}
 	}
 
-	//--------------------------------------------------------------------
-
 	/**
 	 * Scans the body of the request message and replaces any nonce
 	 * placeholders with actual nonces, that we'll then add to our
 	 * headers.
 	 *
-	 * @param ResponseInterface|Response $response
+	 * @param ResponseInterface $response
+	 *
+	 * @return void
 	 */
 	protected function generateNonces(ResponseInterface &$response)
 	{
+		/**
+		 * @var Response $response
+		 */
```

Comment:
Why is this necessary over ResponseInterface? If someone is using their own Response class then this still needs to work.

Reply:
I just put it there so that vscode will not turn red everytime I open this file. The current `ResponseInterface` does not have the method used there but is found in `Response`.

Reply:
Hmm that seems to me a larger issue, that code framework code could rely on an incomplete interface. I would be in favor of considering the current ResponseInterface bugged by not accounting for `getBody()` and thus adding it. Thoughts for another Issue perhaps.

Reply:
OK, I'll revert that for now.